blockchain/state_manager/src/call.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::errors::*;
+use crate::StateManager;
+use async_log::span;
+use blockstore::BlockStore;
+use blockstore::BufferedBlockStore;
+use cid::Cid;
+use clock::ChainEpoch;
+use forest_blocks::Tipset;
+use interpreter::{ApplyRet, ChainRand, DefaultSyscalls, VM};
+use log::trace;
+use log::warn;
+use message::{Message, MessageReceipt, UnsignedMessage};
+
+pub struct InvocResult<Msg>
+where
+    Msg: Message,
+{
+    pub msg: Msg,
+    pub msg_rct: MessageReceipt,
+    pub actor_error: Option<String>,
+}
+
+pub type StateCallResult<T> = Result<InvocResult<T>, Error>;
+
+fn state_call_raw<DB>(
+    state_manager: &StateManager<DB>,
+    msg: &mut UnsignedMessage,
+    bstate: &Cid,
+    rand: &ChainRand,
+    bheight: &ChainEpoch,
+) -> StateCallResult<UnsignedMessage>
+where
```

Comment:
I'll have a deeper look at this later, but what is the point in having state_call and state_call_raw?

Reply:
I don't think it is being called anywhere but I remember someone suggesting a do something similar in the past just to keep it consistent